Consul/Agent.cs

```diff
@@ -751,14 +742,26 @@ public Task<WriteResult> DisableNodeMaintenance(CancellationToken ct = default(C
         /// Providing a CancellationToken can be used to close the connection and stop the
         /// log stream, otherwise the log stream will time out based on the HTTP Client's timeout value.
         /// </summary>
-        public async Task<LogStream> Monitor(LogLevel level = default(LogLevel), CancellationToken ct = default(CancellationToken))
+        public async Task<LogStream> Monitor(LogLevel level = default, bool logJSON = false, CancellationToken ct = default)
         {
             var req = _client.Get<Stream>("/v1/agent/monitor");
             req.Params["loglevel"] = level.ToString().ToLowerInvariant();
+
+            if (logJSON)
+            {
+                req.Params["logjson"] = "true";
+            }
+
             var res = await req.ExecuteStreaming(ct).ConfigureAwait(false);
             return new LogStream(res.Response);
         }
 
+        // MonitorJSON is like Monitor except it returns logs in JSON format.
+        public async Task<LogStream> MonitorJSON(LogLevel level = default, CancellationToken ct = default)
```

Comment:
I like this. It is more obvious what it does than just a simple boolean parameter.✔️

Reply:
I think comment was misleading a little bit, sorry about that. I didn't mean to say that we need to resign entirely from the `myMonitor(LogLevel level = default, bool logJSON = false, CancellationToken ct = default)` overload. I think it is useful, but I also think that `MonitorJSON (LogLevel level = default, CancellationToken ct = default)` is useful as well.

Reply:
I removed the overload on my own accord. After thinking about it, I can't really justify having an overload for a single boolean flag.. I think it might confuse users to choose between using the helper method or the overload..

Reply:
> I just don't like a little bit the code duplication we have at the moment

I feel the same way. I think if we have another extension to the method, we will do that.